File: TinkoffChat/TinkoffChat/ThemesViewController.m
//
//  ThemesViewController.m
//  TinkoffChat
//
//  Created by Алексей ]Чанов on 02/03/2019.
//  Copyright © 2019 Алексей Чанов. All rights reserved.
//

#import "ThemesViewController.h"

@interface ThemesViewController () {
    id<ThemesViewControllerDelegate> _delegate;
    Themes *_model;
}

@end

@implementation ThemesViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.model = [Themes new];
}

- (IBAction)closeTap:(id)sender {
    [self.presentingViewController dismissViewControllerAnimated:YES
                                                      completion:nil];
}

- (void)applyColor:(UIColor *)color {
    if ([self.delegate
         respondsToSelector:@selector(themesViewController:didSelectTheme:)])
    {
        [self.delegate themesViewController:self didSelectTheme:color];
    }
    self.view.backgroundColor = color;
}

- (IBAction)theme1Tap:(id)sender {
    [self applyColor:self.model.theme1];
}

- (IBAction)theme2Tap:(id)sender {
    [self applyColor:self.model.theme2];
}

- (IBAction)theme3Tap:(id)sender {
    [self applyColor:self.model.theme3];
}

- (id<ThemesViewControllerDelegate>)delegate {
    return _delegate;
}

- (void)setDelegate:(id<ThemesViewControllerDelegate>)delegate {
    _delegate = delegate;
}

- (Themes *)model {
    return _model;
}

- (void)setModel:(Themes *)model {
    Themes *tmpModel = _model;
    _model = [model retain];
    [tmpModel release];
}
@end

File: TinkoffChat/TinkoffChat/ThemesViewController.swift
//
//  ThemesViewController.swift
//  TinkoffChat
//
//  Created by Алексей ]Чанов on 05/03/2019.
//  Copyright © 2019 Алексей Чанов. All rights reserved.
//

import UIKit

typealias ThemesClosure = (_ color: UIColor) -> (Void)

class ThemesViewController: UIViewController {
    @IBOutlet var clouseButtonOutlet: UIButton!
    var model: Themes?
    var themesClosure: ThemesClosure?
    
    override func viewDidLoad() {
        super.viewDidLoad()
        model = Themes()
        
        theme1Outlet.layer.cornerRadius = theme1Outlet.frame.size.height*0.2
        theme2Outlet.layer.cornerRadius = theme2Outlet.frame.size.height*0.2
        theme3Outlet.layer.cornerRadius = theme3Outlet.frame.size.height*0.2
        clouseButtonOutlet.layer.cornerRadius = clouseButtonOutlet.frame.size.height*0.2
    }
    
    func applyColor(_ color: UIColor?) {
        guard let color = color else { return }
        
        view.backgroundColor = color
        themesClosure?(color)
        AppDelegate.changeAppearance(color)
    }
    
    @IBAction func theme1Tap(_ sender: UIButton) {
        applyColor(model?.theme1)
    }
    @IBOutlet var theme1Outlet: UIButton!
    
    @IBAction func theme2Tap(_ sender: UIButton) {
        applyColor(model?.theme2)
    }
  
    @IBOutlet var theme2Outlet: UIButton!
    
    @IBAction func theme3Tap(_ sender: UIButton) {
        applyColor(model?.theme3)
    }
    @IBOutlet var theme3Outlet: UIButton!
    
    @IBAction func closeTap(_ sender: UIButton) {
        presentingViewController?.dismiss(animated: true)
    }
}
